acai-ios/SectionControllers/OrderCustomizationListSectionController.swift

```diff
@@ -0,0 +1,103 @@
+    var selectedBaseIndex: Int = 0
+    
+    // MARK: Constraint Constants
+    let baseCellHeightConstraint: CGFloat = 87
+    let toppingCellHeightConstraint: CGFloat = 55
+    
+    init(options: OrderCustomizationOptions) {
+        customizationOptions = options
+    }
+    
+    override func numberOfItems() -> Int {
+        return customizationOptions.options.count
+    }
+    
+    override func sizeForItem(at index: Int) -> CGSize {
+        guard let context = collectionContext else {
+            return .zero
+        }
+        let width = context.containerSize.width
+        switch customizationOptions.options[index].type {
+        case .base?:
+            return CGSize(width: width, height: baseCellHeightConstraint)
+        case .topping?:
+            return CGSize(width: width, height: toppingCellHeightConstraint)
+        case .none:
+            return .zero
+        }
+    }
+    
+    override func cellForItem(at index: Int) -> UICollectionViewCell {
+        var cellType: AnyClass = UICollectionViewCell.self
+        let option = customizationOptions.options[index]
+        switch option.type {
+        case .base?:
+            cellType = RadioSelectionCollectionViewCell.self
+            if option.isSelected {
+                selectedBaseIndex = index
+            }
+        case .topping?:
+            cellType = MultiSelectionCollectionViewCell.self
+        case .none:
+            cellType = UICollectionViewCell.self
+        }
+        
+        let cell = collectionContext!.dequeueReusableCell(of: cellType, for: self, at: index)
+        if let cell = cell as? RadioSelectionCollectionViewCell {
+            cell.customizationOption = option
+            cell.setCustomizationOptionLabelText()
+            cell.updateRadioSelection()
+        } else if let cell = cell as? MultiSelectionCollectionViewCell {
+            cell.customizationOption = option
+            cell.setOrderCustomizationOptionTitleLabelText()
+            cell.updateSelectionRect()
+        }
+        return cell
+    }
+    
+    override func didUpdate(to object: Any) {
+        guard let object = object as? OrderCustomizationOptions else { return }
+        customizationOptions = object
+    }
+    
+    func deselectItem(at index: Int) {
+        guard let type = customizationOptions.options[index].type else { return }
+        switch type {
+        case .base:
+            self.selectOptionDelegate?.deselectOption(at: index)
+        case .topping:
+            return
+        }
+    }
+    
+    override func didSelectItem(at index: Int) {
+        guard let type = customizationOptions.options[index].type else { return }
+        switch type {
+        case .base:
+            deselectItem(at: selectedBaseIndex)
+            self.selectOptionDelegate?.selectOption(at: index, type: type)
+        case .topping:
+            self.selectOptionDelegate?.selectOption(at: index, type: type)
```

Comment:
if we call `self.selectOptionDelegate?.selectOption(at: index, type: type)` for both cases, lets just move it to after the `switch` statement

Reply:
I need something in my case topping though

Reply:
oh you can do 
```
if type == .base {
  deselectItem(...)
}
self.selectOptionDelegate...
```